directory/directory_dest.go

```diff
@@ -4,19 +4,64 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
+	"path/filepath"
 
 	"github.com/containers/image/types"
 	"github.com/opencontainers/go-digest"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 )
 
+const version = "Directory Transport Version: 1.0"
+
 type dirImageDestination struct {
-	ref dirReference
+	ref      dirReference
+	compress bool
 }
 
-// newImageDestination returns an ImageDestination for writing to an existing directory.
-func newImageDestination(ref dirReference) types.ImageDestination {
-	return &dirImageDestination{ref}
+// newImageDestination returns an ImageDestination for writing to a directory.
+func newImageDestination(ref dirReference, compress bool) (types.ImageDestination, error) {
+	d := &dirImageDestination{ref: ref, compress: compress}
+
+	// If directory exists check if it is empty
+	// if not empty, check whether the contents match that of a container image directory and overwrite the contents
+	// if the contents don't match throw an error
+	dirExists, err := pathExists(d.ref.resolvedPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error checking for path %q", d.ref.resolvedPath)
+	}
+	if dirExists {
+		isEmpty, err := isDirEmpty(d.ref.resolvedPath)
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = os.Stat(filepath.Join(d.ref.resolvedPath, "version"))
+
+		if !isEmpty && err != nil && os.IsNotExist(err) {
+			return nil, errors.Wrapf(err, "%q is not empty and not a container image directory. Don't want to overwrite important data, please specifiy a different directory path", d.ref.path)
+		}
+		if !isEmpty && err != nil {
+			return nil, errors.Wrapf(err, "error reading directory %q", d.ref.resolvedPath)
+		}
```

Comment:
`if !isEmpty`, and the `version` file exists, this should also check that it has exactly the expected contents.

(In the future we can add version parsing and compatibility logic here; for now, a straight equality amounts to the same thing.)

Reply:
fixed